pkg/kv/kvserver/below_raft_protos_test.go

```diff
@@ -98,81 +69,55 @@ var belowRaftGoldenProtos = map[reflect.Type]fixture{
+			// used.
 			return roachpb.NewPopulatedInternalTimeSeriesData(r, false)
 		},
-		emptySum:     5531676819244041709,
-		populatedSum: 17471291891947207032,
-	},
-	reflect.TypeOf(&enginepb.MVCCMetadataSubsetForMergeSerialization{}): {
-		populatedConstructor: func(r *rand.Rand) protoutil.Message {
+		func(r *rand.Rand) protoutil.Message {
 			m := enginepb.NewPopulatedMVCCMetadataSubsetForMergeSerialization(r, false)
 			if m.MergeTimestamp != nil {
 				m.MergeTimestamp.Synthetic = nil // never populated below Raft
 			}
 			return m
 		},
-		emptySum:     14695981039346656037,
-		populatedSum: 1187861800212570275,
-	},
-	reflect.TypeOf(&roachpb.RaftReplicaID{}): {
-		populatedConstructor: func(r *rand.Rand) protoutil.Message {
+		func(r *rand.Rand) protoutil.Message {
 			return roachpb.NewPopulatedRaftReplicaID(r, false)
 		},
-		emptySum:     598336668751268149,
-		populatedSum: 9313101058286450988,
-	},
-}
-
-func TestBelowRaftProtos(t *testing.T) {
-	defer leaktest.AfterTest(t)()
-	defer log.Scope(t).Close(t)
-
-	// Enable the additional checks in TestMain. NB: running this test by itself
-	// will fail those extra checks - such failures are safe to ignore, so long
-	// as this test passes when run with the entire package's tests.
-	verifyBelowRaftProtos = true
-
-	slice := make([]byte, 1<<20)
-	for typ, fix := range belowRaftGoldenProtos {
-		if b, err := protoutil.Marshal(reflect.New(typ.Elem()).Interface().(protoutil.Message)); err != nil {
-			t.Fatal(err)
-		} else if err := verifyHash(b, fix.emptySum); err != nil {
-			t.Errorf("%s (empty): %s\n", typ, err)
-		}
+	}
 
-		randGen := rand.New(rand.NewSource(goldenSeed))
+	// An arbitrary number chosen to seed the PRNGs used to populate the tested
+	// protos.
+	const goldenSeed = 1337
+	// We'll randomly populate, marshal, and hash each proto. Doing this more than
+	// once is necessary to make it very likely that all fields will be nonzero at
+	// some point.
+	const itersPerProto = 50
 
-		bytes := slice
-		numBytes := 0
-		for i := 0; i < itersPerProto; i++ {
-			if n, err := fix.populatedConstructor(randGen).MarshalTo(bytes); err != nil {
-				t.Fatal(err)
-			} else {
-				bytes = bytes[n:]
-				numBytes += n
+	w := echotest.NewWalker(t, testutils.TestDataPath(t, t.Name()))
+	for _, fn := range testCases {
+		name := fmt.Sprintf("%T", fn(rand.New(rand.NewSource(0))))
+		name = regexp.MustCompile(`.*\.`).ReplaceAllString(name, "")
+		t.Run(name, w.Run(t, name, func(t *testing.T) string {
+			rng := rand.New(rand.NewSource(goldenSeed))
+			hash := fnv.New64a()
+			for i := 0; i < itersPerProto; i++ {
+				msg := fn(rng)
+				dst := make([]byte, msg.Size())
+				_, err := msg.MarshalTo(dst)
+				require.NoError(t, err)
+				hash.Write(dst)
 			}
-		}
-		if err := verifyHash(slice[:numBytes], fix.populatedSum); err != nil {
-			t.Errorf("%s (populated): %s\n", typ, err)
-		}
+			return fmt.Sprint(hash.Sum64())
```

Comment:
nit: consider dumping the string representation of the Protobuf too, for inspection.

Reply:
We iterate 20x so this isn't so easy, besides the auto-generated stuff tends to be an eyesore. I think there isn't a ton of benefit here. If this test fails it will be on a commit in which there is a corresponding edit to the `.proto` file.

Reply:
Sure, no biggie at all.